### PR TITLE
add flask skeleton

### PR DESCRIPTION
**Pull request by nonococoleo**
# Pull Request Template

flask skeleton added in this pull request as well as test cases. only index and health check api are created.

## Checklist:

- [x] I have written test cases for this change, and my code passed the test
- [x] I have updated documentation to reflect this change
- [x] I have attached this PR to an issue/user story

### CODE REVIEW

File: service/__init__.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# Copyright 2016, 2019 John J. Rofrano. All Rights Reserved.
```
Comment (iLtc) [inline service/__init__.py:1]: We don't need to include the professor's name here.

File: service/routes.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        handler = logging.StreamHandler(sys.stdout)
+        handler.setFormatter(logging.Formatter(fmt))
+        handler.setLevel(log_level)
+        # Remove the Flask default handlers and use our own
+        handler_list = list(app.logger.handlers)
+        for log_handler in handler_list:
+            app.logger.removeHandler(log_handler)
+        app.logger.addHandler(handler)
+        app.logger.setLevel(log_level)
+        app.logger.info('Logging handler established')
```
Comment (iLtc) [inline service/routes.py:89]: Does this part belong to #6?

File: service/routes.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+from . import app
+
+
+######################################################################
+# GET HEALTH CHECK
+######################################################################
+@app.route('/healthcheck')
+def healthcheck():
+    """ Let them know our heart is still beating """
+    return make_response(jsonify(status=200, message='Healthy'), status.HTTP_200_OK)
```
Comment (iLtc) [inline service/routes.py:39]: Please add a test case for this part.

File: service/routes.py [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    """ Checks that the media type is correct """
+    if 'Content-Type' not in request.headers:
+        app.logger.error('No Content-Type specified.')
+        abort(status.HTTP_415_UNSUPPORTED_MEDIA_TYPE, 'Content-Type must be {}'.format(content_type))
+
+    if request.headers['Content-Type'] == content_type:
+        return
+
+    app.logger.error('Invalid Content-Type: %s', request.headers['Content-Type'])
+    abort(status.HTTP_415_UNSUPPORTED_MEDIA_TYPE, 'Content-Type must be {}'.format(content_type))
```
Comment (iLtc) [inline service/routes.py:67]: Please add a test case for this part.

**Review by iLtc — CHANGES_REQUESTED** [review]

File: service/__init__.py [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+# BEFORE you import modules that depend on it !!!
+
+# Create the Flask aoo
+app = Flask(__name__)
+
+# Load Configurations
+app.config.from_object('config')
+
+# Import the routes After the Flask app is created
+from service import routes
```
Comment (KHafner) [inline service/__init__.py:32]: Does models need to be imported from service here as well?

Comment (nonococoleo) [reply service/__init__.py:32]: it could be another story, this one is a pure flask skeleton

File: service/routes.py (hunk 2 above)
Comment (nonococoleo) [reply service/routes.py:89]: remove unnecessage codes

File: service/__init__.py (hunk 1 above)
Comment (nonococoleo) [reply service/__init__.py:1]: cleanup

File: service/routes.py (hunk 3 above)
Comment (nonococoleo) [reply service/routes.py:39]: added

**Review by KHafner — APPROVED** [review]

**Review by iLtc — APPROVED** [review]
LGTM, Thanks!

**Pull request merged**